react/features/base/participants/reducer.js

```diff
@@ -147,31 +130,63 @@ function _participant(state: Object = {}, action) {
 }
 
 /**
- * Listen for actions which add, remove, or update the set of participants in
- * the conference.
+ * Reduces a specific redux action of type {@link PARTICIPANT_JOINED} in the
+ * feature base/participants.
  *
- * @param {Participant[]} state - List of participants to be modified.
- * @param {Object} action - Action object.
- * @param {string} action.type - Type of action.
- * @param {Participant} action.participant - Information about participant to be
- * added/removed/modified.
- * @returns {Participant[]}
+ * @param {Object} state - The participant element of the redux state of the
+ * feature base/participants. Given that the redux action is of type
+ * {@code PARTICIPANT_JOINED}, it is expected that {@code state} is an empty
+ * object.
+ * @param {Action} action - The redux action of type {@code PARTICIPANT_JOINED}
+ * to reduce.
+ * @private
+ * @returns {Object} The next/new participant element of the redux state of the
+ * feature base/participants after the reduction of the specified
+ * {@code action}.
  */
-ReducerRegistry.register('features/base/participants', (state = [], action) => {
-    switch (action.type) {
-    case DOMINANT_SPEAKER_CHANGED:
-    case PARTICIPANT_ID_CHANGED:
-    case PARTICIPANT_UPDATED:
-    case PIN_PARTICIPANT:
-        return state.map(p => _participant(p, action));
-
-    case PARTICIPANT_JOINED:
-        return [ ...state, _participant(undefined, action) ];
+function _participantJoined(state, { participant }) {
```

Comment:
state is unused, let's get rid of it?

Reply:
OK. Will come in "Associate remote participant w/ JitsiConference (_JOINED)".